internal/driver/onvifdiscovery.go

```diff
@@ -191,7 +191,8 @@ func executeRawProbe(conn net.Conn, params netscan.Params) ([]onvif.Device, erro
 	buf := make([]byte, bufSize)
 	// keep reading from the PacketConn until the read deadline expires or an error occurs
 	for {
-		n, _, err := (conn.(net.PacketConn)).ReadFrom(buf)
+		//n, _, err := (conn.(net.PacketConn)).ReadFrom(buf) //comment by edgego
+		n, err := conn.Read(buf) //add by edgego
```

Comment:
@edgego does this change allow your Hikvision camera to be discovered? If so, I would remove the commented out code and add a note that conn.Read is used instead of net.PacketConn ReadFrom to fix an issue with UDP responses no being received.

Also, I made this one line change on my local setup and it works fine discovering my Tapo C200 still.